core/util/shodan.py

```diff
@@ -0,0 +1,78 @@
+"""
+OWASP Maryam!
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import re
+import os
+
+class main:
+
+	def __init__(self, framework, q, key, limit=1, count=10):
+		""" google.com search engine
+
+			framework  : core attribute
+			q          : query for search
+			key		   : API key
+			ip		   : query on a given host IP
+			limit      : count of pages
+		"""
+		self.framework = framework
+		self.q = q
+		self.key = key
+		self._pages = ''
+		self.limit = limit
+		self.num = count
+		self.shodan_api = f"https://api.shodan.io/shodan/host/search?key={self.key}&query={self.q}"
+		self._links = []
```

Comment:
you didn't use it so remove it as well

Reply:
Api key is required for accessing searches. How do I remove it?
I will remove limits for sure, thank you. Although count is needed. I will update in sometime.